### DISCUSSION TITLE
Retorno da sefaz nao tem nada

### DISCUSSION THREAD

**Discussion by valdecirsanches** [TadaSoftware/PyNFe#262] [General]
Com toda certeza tem erro mas a sefaz nao retorna so informa que o lote esta ok 103

<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
<soap:Body>
<nfeResultMsg xmlns="http://www.portalfiscal.inf.br/nfe/wsdl/NFeAutorizacao4">
<retEnviNFe xmlns="http://www.portalfiscal.inf.br/nfe" versao="4.00">
<tpAmb>1</tpAmb>
<verAplic>SP_NFE_PL009_V4</verAplic>
<cStat>103</cStat>
<xMotivo>Lote recebido com sucesso</xMotivo>
<cUF>35</cUF>
<dhRecbto>2023-03-08T22:49:42-03:00</dhRecbto>
<infRec>
<nRec>351010857309423</nRec>
<tMed>1</tMed>
</infRec>
</retEnviNFe>
</nfeResultMsg>
</soap:Body>
</soap:Envelope>

**Comment by valdecirsanches** [comment]
Alguem teria um bom exemplo funcionando para sanar duvidas estou usando o exemplo ja exixtente
    if envio[0] == 0:
        print('Sucesso!')
        print(etree.tostring(envio[1], encoding="unicode").replace('\n', '').replace('ns0:', '').replace(':ns0', ''))

        # em caso de erro o retorno será o xml de resposta da SEFAZ + NF-e enviada
        id = ret.get('NFe').get('NFe').get('infNFe').get('@Id')
        pasta = f'{BASE_DIR}/media/{emp.id}/nfe/autorizadas'
        CriaPasta(pasta)
        xmlname = f'{pasta}/{id[3:]}-nf.xml'
        arquivo = open(xmlname, 'wb')
        arquivo.write(str(xml).encode())
        arquivo.close()

        nf.xmlfile=xmlname
        nf.status='AUTORIZADA'
        nf.save()

        # DanfeRetrato(xmlname)

        return JsonResponse(dict(success=ret), safe=False)
    else:
        # print('Erro:')
        # print(envio[1].text)
        # print('Nota:')
        # print(etree.tostring(envio[2], encoding="unicode"))

        obj = etree.tostring(envio[2], encoding="unicode")
        xml = obj
        obj = xmltodict.parse(obj)
        ret = json.dumps(obj)
        ret = json.loads(ret)
        id = ret.get('NFe').get('infNFe').get('@Id')
        pasta = f'{BASE_DIR}/media/{emp.id}/nfe/rejeitadas'
        CriaPasta(pasta)

        xmlname = f'{pasta}/{id[3:]}-nfe.xml'
        arquivo = open(xmlname, 'wb')
        arquivo.write(str(xml).encode())
        arquivo.close()

        nf.xml_file = xmlname
        nf.status = 'REJEITADA'
        nf.save()

        err = envio[1].text
        xmlname = f'{pasta}/{id[3:]}-erro.xml'
        arquivo = open(xmlname, 'wb')
        arquivo.write(str(err).encode())
        arquivo.close()



        # DanfeRetrato(xmlname)

        return JsonResponse(dict(erro='NFe não autorizada junto a SEFAZ'), safe=False)

**Comment by leogregianin** [comment]
@valdecirsanches
1. A transmissão da nota é para São Paulo, suponho (já que não passou detalhes)
2. Para São Paulo a transmissão da nota é assíncrono, ou seja, você envia o lote e a SEFAZ retorna o recibo e não o status efetivo do processamento da nota.
3. Com o número do recibo, você tem que fazer uma nota requisição a SEFAZ para consultar o recibo e visualizar como resultou o processamento da nota.

Sugiro ler todas as notas técnicas antes de tudo. 
Nesse link tem todas elas: https://www.nfe.fazenda.gov.br/portal/listaConteudo.aspx?tipoConteudo=04BIflQt1aY=

**Reply by valdecirsanches** [reply]
Obrigado amigão vou verificar então e posto aqui o resultado

**Reply by valdecirsanches** [reply]
Agora sim!!!
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
<soap:Body>
<nfeResultMsg xmlns="http://www.portalfiscal.inf.br/nfe/wsdl/NFeRetAutorizacao4">
<retConsReciNFe xmlns="http://www.portalfiscal.inf.br/nfe" versao="4.00">
<tpAmb>2</tpAmb>
<verAplic>SP_NFE_PL009_V4</verAplic>
<nRec>351000173460853</nRec>
<cStat>225</cStat>
<xMotivo>Rejeição: Falha no Schema XML do lote de NFe</xMotivo>
<cUF>35</cUF>
<dhRecbto>2023-03-09T13:23:58-03:00</dhRecbto>
</retConsReciNFe>
</nfeResultMsg>
</soap:Body>
</soap:Envelope>
Sabe me dizer se tem historico deste erro????

**Reply by leogregianin** [reply]
Faça a validação nesse site https://dfe-portal.svrs.rs.gov.br/Mdfe/ValidadorXML do conteúdo do XML que você gerou, nele irá constar o que há de "falha no schema XML"

**Reply by valdecirsanches** [reply]
Parece que nao tive sorte:
![image](https://user-images.githubusercontent.com/17271010/224129360-94966e96-59f5-47aa-9359-9650676b71cc.png)

**Reply by valdecirsanches** [reply]
Estou chegando lá.... sabe me dizer porque o numero_nf nao fez parte da montagem da chave???
![image](https://user-images.githubusercontent.com/17271010/224146167-51dd5d20-f7d7-4f39-9b30-646c4a0998b8.png)

![image](https://user-images.githubusercontent.com/17271010/224146628-d3a94d9a-b91e-49f3-bed4-610452080f3e.png)

**Reply by leogregianin** [reply]
A chave de acesso é gerada automaticamente pelo PyNFe. Em algum momento você mesmo está gerando ela?

**Reply by valdecirsanches** [reply]
Estou verificando

**Reply by valdecirsanches** [reply]
Verificado 
35230330511823000142550010000001581160909681  <-- Rejeição: Erro na Chave de Acesso - (Homologação)
35230330511823000142550010000001581866069025  <-- Aprovado anteriormente (Produção)
Me ajuda a resolver já verifiquei tudo.... creio que so falta isto

![image](https://user-images.githubusercontent.com/17271010/224190944-867d1e0e-1d51-4fdd-a494-b629aeb00e4e.png)

**Reply by valdecirsanches** [reply]
Resolvido
O problema era nos codigos
Conferi o xml e achei a falha!!!
Obrigado.
